### PR TITLE
Port Bay's silicon law manager

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Ports Bay's silicon law manager: https://github.com/Baystation12/Baystation12/pull/8486. Admins and Malf AI's can use this interface to add, edit and remove laws, whereas regular AI's can use it to view their laws and state their laws.
2) Players with antagHUD enabled can now shift-click on AI's and robots to view their laws.

Removes the ability to use a zeroth law card to reset a zeroth law - the purge board will now take care of that. Quarantine is now a priority 8 law again, pending a rewrite into its own core lawset. Also got rid of Crewsimov++ and Antimov++, as no-one ever used those.

Also cleans up a lot of AI code.

Normal silicon view:

Malf/Admin view:

Law sets view:

### CODE REVIEW

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -1,213 +1,289 @@
...
-			var/datum/ai_laws/templaws = new lawtype()
-			inherent = templaws.inherent
-	set_zeroth_law("\red ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*�&110010")
-
-/datum/ai_laws/custom/New() //This reads silicon_laws.txt and allows server hosts to set custom AI starting laws.
-	..()
-	for(var/line in file2list("config/silicon_laws.txt"))
-		if(!line)						continue
-		if(findtextEx(line,"#",1,2))	continue
+var/global/const/base_law_type = /datum/ai_laws/nanotrasen
```
Comment (tigercat2000) [inline code/datums/ai_laws.dm]: Are the per-round-random lawsets still handled somewhere?

Comment (Markolie) [inline code/datums/ai_laws.dm]: make_laws in silicon/laws.dm

I did scrap the whole custom ai_laws.txt system, as that's unused.

File: code/modules/mob/living/silicon/robot/laws.dm [hunk 2]
```diff
@@ -28,6 +29,7 @@
 
 	who << "<b>Obey these laws:</b>"
 	laws.show_laws(who)
+	// TODO: Update to new antagonist system.
```
Comment (tigercat2000) [inline code/modules/mob/living/silicon/robot/laws.dm]: What is with these TODO notes?

Comment (Markolie) [inline code/modules/mob/living/silicon/robot/laws.dm]: Left them in as I intend to port over Bay's antagonist system in the very near future (as it allows for any round to be mixed with any other round: e.g. vampire + nuke ops).

File: code/modules/mob/living/silicon/silicon.dm [hunk 3]
```diff
@@ -3,7 +3,9 @@
 	robot_talk_understand = 1
 	voice_name = "synthesized voice"
 	var/syndicate = 0
-	var/datum/ai_laws/laws = null//Now... THEY ALL CAN ALL HAVE LAWS
+	var/const/MAIN_CHANNEL = "Main Frequency"
+	var/lawchannel = MAIN_CHANNEL // Default channel on which to state laws
```
Comment (tigercat2000) [inline code/modules/mob/living/silicon/silicon.dm]: ... I never realized you could do this via a const var.

**Comment by Robertha89** [comment]
Bay got a multiple gamemode system? Man, I was just considering this as a possibility a few days ago, but that will be sweet, and be anti-meta gaming in nature, I feel i will love it.

**Comment by Markolie** [comment]
Yea, it's pretty cool. It actually makes the game mode code... well... good compared to the shitfest it is right now.

**Comment by TheDZD** [comment]
Can a malf AI remove its 0th law?

**Comment by Markolie** [comment]
No.

**Comment by PsiOmegaDelta** [comment]
You may want to test with a law containing special character, most notably quotations. These would cause issues re-opening the interface (no issues while it was still open) until a recent NanoUI update https://github.com/Baystation12/Baystation12/pull/10873

File: code/modules/mob/living/silicon/robot/laws.dm (hunk 2 above)
Comment (PsiOmegaDelta) [inline code/modules/mob/living/silicon/robot/laws.dm]: While the new antag system is indeed more pleasant to work with than the old variant, it still has some issues, most notably:
* Certain game modes don't end properly, Heist specifically.
* Automatic mid-round antags doesn't currently appear to work.
* Due to the point above Secret is always Extended.
You probably wish for these issues to be fixed first.

**Comment by Zuhayr** [comment]
Don't port the antag system yet, shit's busted.

File: code/modules/mob/living/silicon/robot/laws.dm (hunk 2 above)
Comment (Markolie) [inline code/modules/mob/living/silicon/robot/laws.dm]: That's good to know, thanks. I'll hold off on porting it until those issues are resolved.

**Comment by Markolie** [comment]
Ah yes, I ported the quotations fix when it was merged on Bay. I'll double check anyway though - thanks.

**Pull request merged**